src/internal/components/chart-series-details/index.tsx

```diff
@@ -1,49 +1,117 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
-import React, { memo } from 'react';
+import React, { memo, ReactNode, useEffect, useRef } from 'react';
 import clsx from 'clsx';
 
 import { BaseComponentProps, getBaseProps } from '../../base-component';
-import InternalBox from '../../../box/internal';
-import { ChartDetailPair } from '../../../pie-chart/interfaces';
 import ChartSeriesMarker, { ChartSeriesMarkerType } from '../chart-series-marker';
 import styles from './styles.css.js';
+import InternalExpandableSection from '../../../expandable-section/internal';
+import getSeriesDetailsText from './series-details-text';
+
+interface ChartDetailPair {
+  key: ReactNode;
+  value: ReactNode;
+}
 
 export interface ChartSeriesDetailItem extends ChartDetailPair {
   markerType?: ChartSeriesMarkerType;
   color?: string;
   isDimmed?: boolean;
+  subItems?: ReadonlyArray<ChartDetailPair>;
+  expandable?: boolean;
 }
+export type ExpandedStates = Record<number, boolean>;
 
 export interface ChartSeriesDetailsProps extends BaseComponentProps {
   details: ReadonlyArray<ChartSeriesDetailItem>;
+  expandedStates?: ExpandedStates;
+  setPopoverText?: (s: string) => void;
+  setExpandedState?: (index: number, state: boolean) => void;
 }
 
 export default memo(ChartSeriesDetails);
 
-function ChartSeriesDetails({ details, ...restProps }: ChartSeriesDetailsProps) {
+function ChartSeriesDetails({
+  details,
+  expandedStates,
+  setPopoverText,
+  setExpandedState,
+  ...restProps
+}: ChartSeriesDetailsProps) {
   const baseProps = getBaseProps(restProps);
   const className = clsx(baseProps.className, styles.root);
+  const detailsRef = useRef<HTMLDivElement | null>(null);
+
+  // Once the component has rendered, pass its content in plain text
+  // so that it can be used by screen readers.
+  useEffect(() => {
+    if (setPopoverText) {
+      if (detailsRef.current) {
+        setPopoverText(getSeriesDetailsText(detailsRef.current));
```

Comment:
Almost but it doesn't add commas and screen readers read everything as one sentence without stopping. JAWS, NVDA and VoiceOcer all briefly pause when a comma is found, see [here](https://www.deque.com/blog/dont-screen-readers-read-whats-screen-part-1-punctuation-typographic-symbols/#:~:text=The%20house%20is%20green%2C%20red%2C%20and%20orange.), and since the current approach adds commas between key-value pairs, this would be a regression, even if a small one.

Reply:
Such details need to be incapsulated in shared utility components. There is nothing chart specific, so our utility component should be responsible for that.

Do not add a tech debt by doing a custom version of some typical operation.

Reply:
`getSeriesDetailsText` is chart-specific. On the other hand, I'm thinking that it might make sense to make a separate PR with the changes related to live region from this PR and #1751, and unify how cartesian charts and pie charts generate its content.

Reply:
It might be a general improvement to the live region, not specific to charts. Say, if the passed html includes a list we can separate the entries with commas automatically instead of taking textContent from the entire thing.

Reply:
One problem with lists is that they can be nested, and the `innerText` of the parent already includes the `innerText` of each child.

In order to overcome this here, I am using this specific selector in `getSeriesDetailsText`:

```
`.${styles.announced},.${expandabeleSectionHeaderStyles.header}`
```
where the `styles.announced` class is applied to all key-value pairs except to the ones of top-level items which have expandable sub-items. For these ones, `expandabeleSectionHeaderStyles.header` is used. I can't just add a wrapper around the `ExpandableSection` component because this would also include the children, so I would still have the same problem.

I think one way that such abstraction could work is by either passing either a selector or a list of elements to the function. WDYT?

Reply:
Let's decouple this problem. I created a test page to visualise/test how live region extracts content: https://github.com/cloudscape-design/components/pull/1790

Let's use it with your example and try to get the desired announcement. We can either insert the invisible text (e.g. commas) to the list or modify the `extractInnerText` helper to analyse the nodes and make format the extracted content the way we want.

Reply:
Note that this extracted value is not only used for live region but also for aria labels: [here](https://github.com/cloudscape-design/components/blob/00f52848e1a195893f817f619d0c9995de95e9ae/src/mixed-line-bar-chart/chart-container.tsx#L602) and [here](https://github.com/cloudscape-design/components/blob/00f52848e1a195893f817f619d0c9995de95e9ae/src/mixed-line-bar-chart/chart-container.tsx#L610). So I don't think this can be encapsulated in the live region. What we can do is make this function more generic as mentioned above, and let all charts use it (right now pie and donut charts don't, and use a different approach).